opty/direct_collocation.py

```diff
@@ -166,6 +172,15 @@ def __init__(self, equations_of_motion, state_symbols,
             ndarrays of floats of shape(N,). Any time varying parameters in
             the equations of motion not provided in this dictionary will
             become free trajectories optimization variables.
+        known_experiment_map : dictionary, optional
+            A dictionary that maps the state SymPy functions of time to
+            ndarrays of floats of shape(N,4). Any time varying parameters in
+            the equations of motion not provided in this dictionary will
+            become free trajectories optimization variables.
+	Lambda: float
```

Comment:
This should be an optional flag. If the user supplies the lambda value then it will use the homotopy method and if they don't it should do what the previous behavior was.

Or the default of lambda should be a value that gives the previous behavior (no lambda in the dynamics).

Another approach here would be to make a new Collocator class (subclass this one) and then change out the relevant methods for the homotopy functionality.